readthedocs/config/tests/test_config.py

```diff
@@ -1908,6 +1909,45 @@ def test_search_ranking_normilize_path(self, path, expected):
         build.validate()
         assert build.search.ranking == {expected: 1}
 
+    @pytest.mark.parametrize(
+        'value',
+        [
+            'invalid',
+            True,
+            0,
+            [2, 3],
+            {'foo/bar': 11},
+        ],
+    )
+    def test_search_ignore_invalid_type(self, value):
+        build = self.get_build_config({
+            'search': {'ignore': value},
+        })
+        with raises(InvalidConfig) as excinfo:
+            build.validate()
+        assert excinfo.value.key == 'search.ignore'
+
+    @pytest.mark.parametrize('path, expected', [
+        ('/foo/bar', 'foo/bar'),
+        ('///foo//bar', 'foo/bar'),
+        ('///foo//bar/', 'foo/bar'),
+        ('/foo/bar/../', 'foo'),
+        ('/foo*', 'foo*'),
+        ('/foo/bar/*', 'foo/bar/*'),
+        ('/foo/bar?/*', 'foo/bar?/*'),
```

Comment:
I would make a decision here and support only one case: with or without starting `/`, but not both. Having to values for the same config may only confuse users: `/foo/bar` will work and `foo/bar` too. Both will have the same effect, but I would be expecting different things as a user.

I think it makes more sense to use relative paths here, considering that `/en/latest/` is not taken into account here. What do you think?

Reply:
I don't see that confusing, it would be annoying to fail the build just because you included or not `/`